anchore_engine/services/policy_engine/engine/feeds/grypedb_sync.py

```diff
@@ -119,7 +119,10 @@ def _check_sync_necessary(cls):
         local_grypedb_checksum = cls._get_local_grypedb_checksum()
 
         sync_needed = local_grypedb_checksum != active_grypedb.checksum
-        logger.info("No grypedb sync needed at this time")
+        if not sync_needed:
+            logger.info("No Grype DB sync needed at this time")
+        else:
+            logger.info("Grype DB sync is required.")
 
         return cls.SyncNecessaryResp(sync_needed, active_grypedb)
```

Comment:
Not related to this commit but this violates separation of concerns. If the intent is to check whether a sync is needed, the method should just return a boolean. I see that it's an optmization but the intent does not match the response.

Reply:
fixed